### PR TITLE
SDIT-910 Add role restriction on /api/prisoners/{offenderNo} endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt [hunk 1]
```diff
@@ -141,6 +142,7 @@ class PrisonerResource(private val globalSearchService: GlobalSearchService) {
     ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]),
   )
   @Operation(summary = "List of offenders globally matching the offenderNo.", description = "List of offenders globally matching the offenderNo.")
+  @VerifyOffenderAccess(overrideRoles = ["SYSTEM_USER", "VIEW_PRISONER_DATA"])
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt:145]: still unsure if need GLOBAL_SEARCH.  I initially dismissed this but a similar endpoint (/api/prisoners) endpoint allows multiple offenderNos which can be called instead - and this includes GLOBAL_SEARCH

File: src/test/resources/features/prisoner_search.feature [hunk 2]
```diff
@@ -96,17 +96,14 @@ Feature: Prisoner Search
...
+     Given a user has a token name of "GLOBAL_SEARCH"
     When a search is made for prisoners with offender numbers of "<offenderNos>" using simple endpoint
     Then "<numberResults>" prisoner records are returned
     And the prisoners last names match "<lastNames>"
 
     Examples:
       | offenderNos        | numberResults | lastNames       |
-      | A1234AC            | 1             | BATES           |
-      | A1476AE            | 0             |                 |
-      | A1181MV            | 1             | O'VAUGHAN       |
```
Comment (karenmillermoj) [inline src/test/resources/features/prisoner_search.feature:109]: Removed tests associated to endpoint with new role

File: src/test/resources/features/prisoner_search_with_aliases.feature [hunk 3]
```diff
@@ -89,18 +89,6 @@ Feature: Prisoner Search results contain aliases
       | A1476AE    | 0             |           |
       | A1181MV    | 1             | O'VAUGHAN |
 
-  Scenario Outline: Search for prisoners with specified offender number using simple unprotected endpoint
-    Given a user has a token name of "GLOBAL_SEARCH"
-    When a search is made for prisoners with offender numbers of "<offenderNo>" using simple endpoint
```
Comment (karenmillermoj) [inline src/test/resources/features/prisoner_search_with_aliases.feature:94]: Removed tests associated to endpoint with new role - these already exist in the Resource Test

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt (hunk 1 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt:145]: think it is fine not to include it

File: src/test/resources/features/prisoner_search.feature [hunk 4]
```diff
@@ -96,17 +96,14 @@ Feature: Prisoner Search
       | A1476AE    | 0             |           |
       | A1181MV    | 1             | O'VAUGHAN |
 
-  Scenario Outline: Search for prisoners with specified offender number using simple unprotected endpoint
-    Given a user has a token name of "GLOBAL_SEARCH"
+  Scenario Outline: Search for prisoners with specified offender numbers using simple endpoint
+     Given a user has a token name of "GLOBAL_SEARCH"
```
Comment (petergphillips) [inline src/test/resources/features/prisoner_search.feature:100]: extra space added here

File: src/test/resources/features/prisoner_search.feature (hunk 2 above)
Comment (petergphillips) [reply src/test/resources/features/prisoner_search.feature:109]: if there are no tests now for `/api/prisoners/XXX` can that step definition now be removed?

**Pull request merged**